faust/stores/memory.py

```diff
@@ -31,7 +31,9 @@ def apply_changelog_batch(
             self._create_batch_iterator(to_delete.add, to_key, to_value, batch)
         )
         for key in to_delete:
-            delete_key(key, None)
+            # If the key was assigned a value again at a later time, it will not be deleted.
```

Comment:
Could you add a quick unit test to test this?

Reply:
I have added a test for this case. Until now there was no test for this use case.